pkg/services/notifications/webhook.go

```diff
@@ -22,6 +21,15 @@ type Webhook struct {
 	HttpMethod  string
 	HttpHeader  map[string]string
 	ContentType string
+
+	// Validation is a function that will validate the response body and statusCode of the webhook. Any returned error will cause the webhook request to be considered failed.
+	// This can be useful when a webhook service communicates failures in creative ways, such as using the response body instead of the status code.
+	Validation func(body []byte, statusCode int) error
```

Comment:
I am not sure if this should be `[]byte` or `io.Reader` and we just pass it `resp.Body`?

Reply:
`resp.Body` is consumed on read, we would have to ensure that we reconstruct the `io.Reader` buffer from original data in `sendWebRequestSync` if we passed it into the Validate method.